src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.Android.cs

```diff
@@ -44,16 +44,19 @@ private void SaveTime(TimeSpan time)
 
 		private void AdjustAndSaveTime(int hourOfDay, int minutes)
 		{
+			var oldTime = Time;
 			if (Time.Hours != hourOfDay || Time.Minutes != minutes)
 			{
 				if (_dialog.IsInSpinnerMode)
 				{
 					minutes = minutes * MinuteIncrement;
 				}
 
-				var time = new TimeSpan(Time.Days, hourOfDay, minutes, Time.Seconds, Time.Milliseconds);
+				var time = new TimeSpan(0, hourOfDay, minutes, 0);
```

Comment:
That part is likely to cause some issues. Let's add a feature flag in `FeatureConfiguration` to provide the ability to bring back the original behavior.